operator/pkg/ciliumendpointbatch/reconciler.go

```diff
@@ -0,0 +1,135 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciliumendpointbatch
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	clientset "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2alpha1"
+	"github.com/sirupsen/logrus"
+)
+
+// reconciler is used to sync the current[i.e. desired] state of CEB's in datastore into current state CEB's in k8s-apiserver.
+// The source of truth is in local datastore, client code reconcile with apiserver at regular intervals.
+type reconciler struct {
+	client     clientset.CiliumV2alpha1Interface
+	cebManager cebManager
+}
+
+// Create and Initializes new reconciler.
+func newReconciler(client clientset.CiliumV2alpha1Interface, cebMgr cebManager) *reconciler {
+	return &reconciler{
+		client:     client,
+		cebManager: cebMgr,
+	}
+}
+
+// reconcileWithServer gets the list of ceb's to create, update and delete and syncs with api-server.
+// On error, it returns the list of failed createCebs, updateCebs and deleteCebs to the caller.
+func (r *reconciler) reconcileWithServer(cebsToCreate, cebsToUpdate, cebsToDelete []string) (errCebCreate, errCebUpdate, errCebDelete []string) {
+	var ok, err error
+	var retCeb, ceb *cilium_v2.CiliumEndpointBatch
+
+	// Loop through list of cebsToUpdate and update the CEB
+	for _, c := range cebsToUpdate {
+
+		// Before syncing with ApiServer, get the list of removed CEPs
+		// till from last successful sync with api-server.
+		remCeps := r.cebManager.getRemovedCeps(c)
+		// Get the copy of Ceb from the cebManager
+		if ceb, ok = r.cebManager.getCebFromCache(c); ok != nil {
```

Comment:
Calling the return error `ok` is very confusing, just call it `err`.

Reply:
Its just variable name. By the way, we are using `err` variable inside the code for another check.

Reply:
It's not just variable name, it contains meaning of the variable, if it is an error, just call it `err`, it doesn't matter if inside the function you already called it err, you should, because both are errors.

Reply:
I think, having 2 same variable names in a function it complicates reader.